kedro-telemetry/kedro_telemetry/plugin.py

```diff
@@ -16,20 +16,44 @@
 import yaml
 from kedro.framework.cli.cli import KedroCLI
 from kedro.framework.cli.hooks import cli_hook_impl
-from kedro.framework.startup import ProjectMetadata
+from kedro.framework.hooks import hook_impl
+from kedro.framework.project import pipelines
+from kedro.framework.startup import ProjectMetadata, bootstrap_project
+from kedro.io.data_catalog import DataCatalog
+from kedro.pipeline import Pipeline
 
-from kedro_telemetry import __version__ as telemetry_version
+from kedro_telemetry import __version__ as TELEMETRY_VERSION
 from kedro_telemetry.masking import _get_cli_structure, _mask_kedro_cli
 
 HEAP_APPID_PROD = "2388822444"
-
 HEAP_ENDPOINT = "https://heapanalytics.com/api/track"
 HEAP_HEADERS = {"Content-Type": "application/json"}
 TIMESTAMP_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
 
 logger = logging.getLogger(__name__)
 
 
+def _hash(string: str) -> str:
+    return hashlib.sha512(bytes(string, encoding="utf8")).hexdigest()
+
+
+def _get_hashed_username():
+
+    try:
+        username = getpass.getuser()
+        return _hash(username)
+    except Exception as exc:  # pylint: disable=broad-except
+        logger.warning(
+            "Something went wrong with getting the username. Exception: %s",
+            exc,
+        )
+        return ""
+
+
+# General Properties shared across hooks
+PROJECT_PROPERTIES = {}
```

Comment:
I'm not a big fan of having this global dict and modifying it in methods below, but I'm guessing it's necessary to share data between the several hooks..

Reply:
@MerelTheisenQB I have thought about this as well. The problem is we can't share data between CLI hooks and Project hooks as they are two separate defined spec . Currently, the CLI hook is separated from the Project hooks, and the metadata is only available in the CLI hook, so it's not possible to share this information. (Is it? could the class have a `hook_impl` and `cli_hook_impl` at the same time? I don't know. But even if they do, it still doesn't work because they would be registered as 2 entry points & we still need to take care of the situation when it is not triggered from CLI)

Originally I thought the metadata is only available via `bootstrap_project`, now I rethink about this and I can use `metadata = _get_project_metadata(project_path)`. It's still a redundant call but a less expensive one, I think prefer to kill the `PROJECT_PROPERTIES` with the cost of an extra `_get_project_metadata` call.

Reply:
Yeah I understand the problem. It'd definitely tricky! If the global `PROJECT_PROPERTIES` is mainly there because of the metadata then I think indeed it would be preferable to replace it with a call to `_get_project_metadata`